src/core/components/Search/SearchForm.js

```diff
@@ -0,0 +1,64 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { loadEntities } from 'core/actions';
+import { fetchAddon } from 'core/api';
+import { gettext as _ } from 'core/utils';
+
+import 'core/css/SearchForm.scss';
+import 'core/css/lib/buttons.scss';
+
+export class SearchFormBase extends React.Component {
+  static propTypes = {
+    pathname: PropTypes.string.isRequired,
+    query: PropTypes.string.isRequired,
+  }
+  static contextTypes = {
+    router: PropTypes.object,
+  }
+
+  goToSearch(query) {
+    const { pathname } = this.props;
+    this.context.router.push(`${pathname}?q=${query}`);
+  }
+
+  handleSearch = (e) => {
+    this.goToSearch(this.refs.query.value);
+    e.preventDefault();
```

Comment:
I'm not sure if it makes any difference but it feels to me like this should be first.